### PR TITLE
feat: menu component

### PR DESCRIPTION
**Pull request by kylemcd**
### Description
- Utilizes Radix's menu component
- Adds `shadow` support to `<Box/>` component
- Fixes ordering of `RefToTgphRef` props to avoid collisions

### Tasks
[KNO-6142](https://linear.app/knock/issue/KNO-6142/[telegraph]-menu-component)

### CODE REVIEW

**Comment by kylemcd** [comment]
* **#132** <a href="https://app.graphite.dev/github/pr/knocklabs/telegraph/132?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @kylemcd and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/menu/src/Menu/Menu.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  args: {},
+};
+
+export default meta;
+
+type Story = StoryObj<TgphComponentProps<typeof TelegraphMenu.Root>>;
+
+export const Default: Story = {
+  render: ({ ...args }) => {
+    return (
```
Comment (connorlindsey) [inline packages/menu/src/Menu/Menu.stories.tsx:20]: Nit: It'd be nice to have a full example with the menu button as well to open/close it but not necessary to get this shipped unless there are any gotchas there

**Review by connorlindsey — APPROVED** [review]
Looks great!

File: packages/menu/src/Menu/Menu.stories.tsx (hunk 1 above)
Comment (kylemcd) [reply packages/menu/src/Menu/Menu.stories.tsx:20]: That is going to be a separate component `dropdown-menu`, this component is just the menu contents as it'll be used across a few different component types.

Comment (connorlindsey) [reply packages/menu/src/Menu/Menu.stories.tsx:20]: Ahh, makes sense!

**Pull request merged**